src/build.c

```diff
@@ -1081,8 +1082,15 @@ static gchar *build_create_shellscript(const gchar *working_dir, const gchar *cm
 #ifdef G_OS_WIN32
 	/* Expand environment variables like %blah%. */
 	expanded_cmd = win32_expand_environment_variables(cmd);
-	str = g_strdup_printf("cd \"%s\"\n\n%s\n\n%s\ndel \"%%0\"\n\npause\n", working_dir, expanded_cmd, (autoclose) ? "" : "pause");
+	str = g_strdup_printf(
+		"cd \"%s\"\n\n\n%s\n\n%s\ndel \"%%0\"\n\npause\n",
+		working_dir, expanded_cmd, (autoclose) ? "" : "pause");
 	g_free(expanded_cmd);
+	/* Convert script content into system codepage */
+	utf8_script_content = win32_convert_to_system_codepage(str, error);
```

Comment:
@eht16 AFAICT [this line](https://github.com/geany/geany/pull/1095/files#diff-8ae5a6e4e9368d91c0c59edbde1627c2R1077) creates a filename in glib encoding, ie UTF8, but when its used [here](https://github.com/eht16/geany/blob/6135e82b50867bea790ac2e6382d8c4e249685f9/src/build.c#L895) it has not been converted to locale.